application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
Without doing more debugging, I believe the status code remains zero because the non-Okay OPTIONS request causes the request to just close, thereby bypassing code which would have set status code, headers, etc.